core/src/consensus.rs

```diff
@@ -395,6 +469,14 @@ impl Tower {
         false
     }
 
+    fn use_stray_restored_ancestors(
```

Comment:
Admittedly, I extracted this `fn` only for unit-tests: https://github.com/solana-labs/solana/pull/10718/commits/8c65c2ea59050a6992472c49b31d4c4f2fa677c0.

This code-arrangement itself incurs another needless lookup of `HashMap<Slot, HashSet<Slot>>` for each `check_switch_threshold` invocation. But I think this is tolerable.

Reply:
I see your point,  because when this is called in `check_switch_threshold` we already know, `ancestors.contains_key(&slot)`. Since this first check is only really used in tests, can we make this a test only function and replace the call to `use_stray_restored_ancestors()` with just `self.is_stray_last_vote()` below?

Reply:
I see your point,  because when this is called in `check_switch_threshold` we already know `ancestors.contains_key(&slot)`. Since this first check is only really used in tests, can we make this a test only function and replace the call to `use_stray_restored_ancestors()` with just `self.is_stray_last_vote()` (more clarity) below?

Reply:
done!